modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/RoleResourceV2Manager.java

```diff
@@ -806,4 +1052,134 @@ private List<Map<String, String>> transformAttributeToMap(MultiValuedAttribute m
         }
         return memberList;
     }
+
+    private SCIMResponse updateUsersWithPatchOperations(String existingRoleId, List<PatchOperation> opList,
+                                                        RoleV2Manager roleManager, SCIMResourceTypeSchema schema,
+                                                        JSONEncoder encoder) {
+
+        try {
+            Map<String, List<PatchOperation>> patchOperations = new HashMap<>();
```

Comment:
Why these logic are duplicated in `updateGroupsWithPatchOperations`, `updateUsersWithPatchOperations`, and `updateWithPatchOperationsMeta`

Reply:
Moved the logic to a private method: https://github.com/wso2/charon/pull/435/commits/adb417717e375a2b27daa8200ee3b635f49e8332